src/cohort.js

```diff
@@ -0,0 +1,114 @@
+class Cohort {
+  constructor() {
+    this.cohortObject = {
+      cohortName: '',
+      studentList: []
+    }
+    this.cohortList = []
+    this.allStudents = []
+  }
+
+  findCohort(cohortName) {
+    const cohort = this.cohortList.find(
+      (cohort) => cohort.cohortName === cohortName
+    )
+    if (!cohort) {
+      return 'cohort does not exist'
+    }
+    return cohort
+  }
+
+  findStudent(studentID) {
+    const student = this.allStudents.find(
+      (student) => student.studentID === studentID
+    )
+    if (!student) {
+      return 'student does not exist'
+    }
+    return student
+  }
+
+  createCohort(cohortName) {
+    const existingCohort = this.cohortList.find(
+      (cohort) => cohort.cohortName === cohortName
+    )
+    if (existingCohort) {
+      return 'cohort already exists'
+    }
+
+    this.cohortObject.cohortName = cohortName
+    this.cohortList.push(this.cohortObject)
+    return this.cohortList
+  }
+
+  createStudent(uniqueID, firstName, lastName, githubUsername, email) {
+    const newStudent = {
+      studentID: uniqueID,
+      firstName,
+      lastName,
+      githubUsername,
+      email
+    }
+    this.studentID++
+    this.allStudents.push(newStudent)
+    return newStudent
+  }
+
+  addStudentToCohort(studentID, cohortName) {
+    const whichCohort = this.findCohort(cohortName)
+    const whichStudent = this.findStudent(studentID)
+    const existingStudent = whichCohort.studentList.find(
+      (student) => student.studentID === studentID
+    )
+    if (existingStudent) {
+      return 'student already in this cohort'
+    }
+    whichCohort.studentList.push(whichStudent)
+    return this.cohortList
+  }
+
+  removeCohort(cohortName) {
+    const whichCohort = this.findCohort(cohortName)
+    if (whichCohort === 'cohort does not exist') {
+      return 'cohort does not exist'
+    }
```

Comment:
This could be made into one line by .findCohort() throwing an error instead of returning a string

Reply:
I ended up doing it this way to work around findCohort returning the error message string, will have a look at refactoring this!